contracts/Delay.sol

```diff
@@ -148,22 +139,81 @@ contract Delay is AragonApp, IForwarder {
     */
     function forward(bytes _evmCallScript) public {
         require(canForward(msg.sender, _evmCallScript), ERROR_CAN_NOT_FORWARD);
-        _delayExecution(_evmCallScript);
+        _delayExecution(new bytes(0), _evmCallScript);
+    }
+
+    /**
+    * @dev Challenge script execution
+    * @param _delayedScriptId The ID of the script execution to challenge
+    */
+    function _onDisputableChallenged(uint256 _delayedScriptId, address /* _challenger */) internal {
+        _pauseExecution(_delayedScriptId);
+    }
+
+    /**
+    * @dev Allow script execution
+    * @param _delayedScriptId The ID of the script execution to allow
+    */
+    function _onDisputableAllowed(uint256 _delayedScriptId) internal {
+        _resumeExecution(_delayedScriptId);
+    }
+
+    /**
+    * @dev Reject script execution
+    * @param _delayedScriptId The ID of the script execution to reject
+    */
+    function _onDisputableRejected(uint256 _delayedScriptId) internal {
+        require(_isExecutionPaused(_delayedScriptId), ERROR_SCRIPT_NOT_PAUSED);
```

Comment:
Here and implicitly in the other hooks, we have some requires.

I wonder if we shouldn't treat these hooks as "always safe", as I assume reverting there would be really bad (potentially deadlocking that court case?)

Reply:
Interesting question, although not sure what’s best here. Not reverting may be bad too, leaving contracts in inconsistent states.